hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -195,17 +195,24 @@ public Partitioner getUpsertPartitioner(WorkloadProfile profile) {
 
                   // append rollback blocks for updates
                   if (commitMetadata.getPartitionToWriteStats().containsKey(partitionPath)) {
+                    // This needs to be done since GlobalIndex at the moment does not store the latest commit time
```

Comment:
From the diff, I think the goal for this change is to correctly identify the base-commit so that we can append a rollback block. In that case, Can't we just get the base-commit from the writeStat all the time (global index or not ) ? The AppendHandle sets the base-commit as the previous commit in the write-stat. I may be missing some context here.

Reply:
This is the scenario where the write failed, we record old commitTimes in the inflight file.